services/venice-router/src/main/java/com/linkedin/venice/router/api/VeniceVersionFinder.java

```diff
@@ -88,9 +88,13 @@ public int getVersion(String storeName, BasicFullHttpRequest request) throws Ven
 
     int metadataCurrentVersion = store.getCurrentVersion();
     if (!lastCurrentVersionMap.containsKey(storeName)) {
-      lastCurrentVersionMap.put(storeName, metadataCurrentVersion);
+      if (metadataCurrentVersion != Store.NON_EXISTING_VERSION && !isDecompressorReady(store, metadataCurrentVersion)) {
+        return Store.NON_EXISTING_VERSION;
```

Comment:
There is some logic handling decompressor not ready case in `maybeServeNewCurrentVersion` which will be dead code moving forward. can we clean that up as well? Thanks.